docs/names-and-ids.md

```diff
@@ -0,0 +1,43 @@
+# Names and IDs
+
+version .1
+
+There are a number of names and IDs used in THIEF.  This table describes some of them:
+
+| Identifier | format | Meaning |
+| - | - | - |
+| servicePool | string | Name of pool that a service app belongs to. |
+| requestedServicePool | string | Name of pool that a device app would like to pair with. |
+| runId | guid | Guid representing a running executable. |
+| serviceAppRunId | guid | Guid representing a running _service_ executable. |
+| sessionId | guid | Guid representing the pairing between a device app and a service app. |
+| pingbackId | guid | Guid used to represent the ACK of a thief operation. |
+
+## servicePool notes
+* A service pool contains one or more running service apps.
+* Each running service app belongs to exactly one pool.
+* Service pools were added to support a fallback scenario where a device could detect a non-responsive service app and decide to pair with a different service app.
+* Since service fallback isn't implemented, there currently no advantage to having more than one service app in any given pool.
+
+## requestedServicePool notes
+* A device specifies the nam of a service pool that it would like to pair with.
+* The device _must_ specify a requested service pool.
+* Any service app in the requested pool can pair with the device app.
+* A service app cannot pair with a device app that is requesting a different pool.
+* If no service apps are available in the device's requested pool, the pairing fails. 
+
+## runId notes
```

Comment:
We probably should have the device include the device runId in it's messages and twin, so when the service logs, it includes that runId in custom properties. Same for service including its runId in communications so the device can log that it its custom properties.

This way we can find all the associated logs for a device's run (including the relevant service logs).

Thoughts?

Reply:
Yes, we should do that.